tools/code-owners-parser/CodeOwnersParser/CodeownersEntry.cs

```diff
@@ -157,5 +167,24 @@ private static bool IsGitHubUserAlias(string alias)
             }
             return true;
         }
+
+        protected bool Equals(CodeownersEntry other)
+            => PathExpression == other.PathExpression
+               && Owners.SequenceEqual(other.Owners)
+               && PRLabels.SequenceEqual(other.PRLabels)
+               && ServiceLabels.SequenceEqual(other.ServiceLabels);
+
+        public override bool Equals(object? obj)
+        {
+            // @formatter:off
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((CodeownersEntry)obj);
+            // @formatter:on
+        }
+
+        public override int GetHashCode()
+            => HashCode.Combine(PathExpression, Owners, PRLabels, ServiceLabels);
```

Comment:
@weshaggard I am using `SequenceEqual` in `Equals` above, so I do believe this is not exactly right approach to computing hash, per [this Stack Overflow question](https://stackoverflow.com/questions/28546239/handling-collections-in-gethashcode-implementation). However, I am not sure if this is big enough problem to try to get the hash code computed properly.

Reply:
It isn't that critical unless it is used in a lookup table and there is risk of collision.

Reply:
@weshaggard https://github.com/Azure/azure-sdk-tools/pull/5636